Requesting latest featured mods files causes exception

```
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is java.util.concurrent.CompletionException: java.lang.NullPointerException: null value in entry: group=null
     at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:982)
     at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)
     at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
     at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
     at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
     at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841)
     at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
     at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
     at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317)
     at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:116)
     at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:84)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
     at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214)
     at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177)
     at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346)
     at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262)
     at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
     at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
     at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:106)
     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
     at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
     at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
     at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
     at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
     at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
     at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
     at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
     at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
     at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
     at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
     at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
     at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
     at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
     at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
     at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
     at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
     at org.eclipse.jetty.server.Server.handleAsync(Server.java:609)
     at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:334)
     at org.eclipse.jetty.server.HttpChannel.run(HttpChannel.java:264)
     at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:673)
     at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:591)
     at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.CompletionException: java.lang.NullPointerException: null value in entry: group=null
     at org.springframework.aop.interceptor.AsyncExecutionAspectSupport$CompletableFutureDelegate$1.get(AsyncExecutionAspectSupport.java:331)
     at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
     ... 1 common frames omitted
Caused by: java.lang.NullPointerException: null value in entry: group=null
     at com.google.common.collect.CollectPreconditions.checkEntryNotNull(CollectPreconditions.java:34)
     at com.google.common.collect.ImmutableMapEntry.<init>(ImmutableMapEntry.java:49)
     at com.google.common.collect.ImmutableMap.entryOf(ImmutableMap.java:172)
     at com.google.common.collect.ImmutableMap.of(ImmutableMap.java:159)
     at com.faforever.api.featuredmods.FeaturedModsController.lambda$modFileMapper$0(FeaturedModsController.java:58)
     at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
     at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
     at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
     at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
     at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
     at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
     at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
     at com.faforever.api.featuredmods.FeaturedModsController.getFiles(FeaturedModsController.java:51)
     at com.faforever.api.featuredmods.FeaturedModsController$$FastClassBySpringCGLIB$$72755367.invoke(<generated>)
     at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
     at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:738)
     at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
     at org.springframework.aop.interceptor.AsyncExecutionInterceptor$1.call(AsyncExecutionInterceptor.java:115)
     at org.springframework.aop.interceptor.AsyncExecutionAspectSupport$CompletableFutureDelegate$1.get(AsyncExecutionAspectSupport.java:328)
     ... 4 common frames omitted
```

--- a/src/main/java/com/faforever/api/featuredmods/LegacyFeaturedModFileRepository.java
+++ b/src/main/java/com/faforever/api/featuredmods/LegacyFeaturedModFileRepository.java
@@ -48,7 +48,7 @@ public class LegacyFeaturedModFileRepository implements Repository<FeaturedModFi
         "  ) latest" +
         "    ON file.fileId = latest.fileId" +
         "       AND file.version = latest.version" +
-        "  LEFT JOIN updates_%1$s b" +
+        "  INNER JOIN updates_%1$s b" +
         "    ON b.id = file.fileId;", innerModName, (version == null ? "" : "WHERE version <= :version")), FeaturedModFile.class);
 
     if (version != null) {

